### PR TITLE
29 Add date component

### LINKED ISSUE #29: #29 Date Component
Shows Today, Tomorrow or else the date

### PR DESCRIPTION
**Pull request by hananina**
closes https://github.com/chingu-voyages/v25-geckos-team-04/issues/29

## Component is used like this:
`<DateComponent date={new Date()} formatString='eee d MMM' />`

Date attribute, it should take in the current date. 
formatString, it should take in the format string `eee d MMM` https://date-fns.org/v2.16.1/docs/format

------

### When you set tomorrow's date as an argument
スクリーンショット 2020-11-20 23 07 36
スクリーンショット 2020-11-20 23 07 51

### When you set a date other than tomorrow
スクリーンショット 2020-11-20 23 08 14
スクリーンショット 2020-11-20 23 08 08

### CODE REVIEW

**Comment by hananina** [comment]
@mrpeterhanley @inovux I couldn't take a screenshot because when I call this component in App.tsx, the props pass in to the date attribute seems not working...I googled it but can't figure out what is causing this problem😭
please let me know if there is a way to solve this error🙏

スクリーンショット 2020-11-20 16 21 27
スクリーンショット 2020-11-20 16 21 15

**Review by mrpeterhanley — APPROVED** [review]
Sorry, I will re-check your component

**Comment by mrpeterhanley** [comment]
I think because you named your component "Date", then when you try to create a new Date(), it tries to create another copy of your component instead of a Date() object, because they are both the same name

From Stack Overflow:
https://stackoverflow.com/questions/43744312/react-js-get-current-date/43744409
stackoverflow

I changed your component name to "DateComponent" and it works:
DateComponent

**Comment by mrpeterhanley** [comment]
I renamed to DateComponent and fixed some errors. Please check the component now

**Comment by hananina** [comment]
@mrpeterhanley Oh...Name conflict was the cause of that error, It makes sense! Thank you for fixing that🙏🙏

**Comment by hananina** [comment]
スクリーンショット 2020-11-20 23 08 14
now I can see the date on my browser😭🙌

File: src/components/DateComponent/index.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import { format, isTomorrow } from 'date-fns'
+import styles from './Date.module.css'
+
+interface IDateComponent {
+  date: Date
+}
+
+export const DateComponent: FC<IDateComponent> = ({ date }) => {
+  const checkTomorrow = isTomorrow(date)
+  const formattedDate = format(date, 'eee d MMM')
```
Comment (inovux) [inline src/components/DateComponent/index.tsx]: I think we can use the format string `eee d MMM` as a property to the component called `formatString` so we can even make it more reusable. What do you think @hananina ?

**Review by inovux — CHANGES_REQUESTED** [review]

File: src/components/DateComponent/index.tsx (hunk 1 above)
Comment (hananina) [reply src/components/DateComponent/index.tsx]: I think it's a good idea! I added the property to DateComponent🙋🏻‍♀️
<DateComponent date={new Date()} formatString='eee d MMM' /> @inovux

**Review by inovux — APPROVED** [review]
Looks good :)

**Pull request merged**